### PR TITLE
fix: [CDS-41683]: changed path to extend support for initial values in env V2 runtime

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

changed path to extend support for initial values in env V2 runtime

#### Screenshots

https://user-images.githubusercontent.com/106532291/184077902-031cd8b2-b757-49a9-b7a3-cdc21f8c7daa.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10526)

**Comment by vikrant-harness** [comment]
retrigger titlecheck

**Comment by vikrant-harness** [comment]
retrigger typecheck

**Review by Sagar-Rajput7 — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx [hunk 1]
```diff
@@ -969,6 +969,23 @@ export function StageInputSetFormInternal({
                             (deploymentStage?.deploymentType as StepType) || '',
                             getString
                           )}
+                          onUpdate={data => {
+                            /* istanbul ignore next */
+                            if (
+                              deploymentStageInputSet?.environment?.infrastructureDefinitions?.[index]?.inputs?.spec
+                            ) {
+                              deploymentStageInputSet.environment.infrastructureDefinitions[index].inputs.spec = data
```
Comment (YounixIO) [inline src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:977]: safety check

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx [hunk 2]
```diff
@@ -969,6 +969,23 @@ export function StageInputSetFormInternal({
                             (deploymentStage?.deploymentType as StepType) || '',
                             getString
                           )}
+                          onUpdate={data => {
+                            /* istanbul ignore next */
+                            if (
+                              deploymentStageInputSet?.environment?.infrastructureDefinitions?.[index]?.inputs?.spec
```
Comment (YounixIO) [inline src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:975]: Destructure it to use a one word variable.

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:977]: Safety check is applied on the above line , safety checks cannot be applied while assignment :-> Typescript throws error

Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:977]: Safety check is applied on the above line , safety checks cannot be applied while assignment :-> Typescript throws error

File: src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/StageInputSetForm.tsx:975]: Done this in latest commit

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**